mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxViewModelProvider.kt

```diff
@@ -87,19 +87,14 @@ object MvRxViewModelProvider {
 }
 
 /**
- * Return the [Class] of the [MvRxViewModelFactory] for a given ViewModel class, if it exists.
+ * Return the [Class] of the companion [MvRxViewModelFactory] for a given ViewModel class, if it exists.
  */
 internal fun <VM : BaseMvRxViewModel<*>> Class<VM>.factoryCompanion(): Class<out MvRxViewModelFactory<VM, *>>? {
-    val companionClass = try {
-        Class.forName("$name\$Companion")
-    } catch (exception: ClassNotFoundException) {
-        return null
-    }
-    return if (MvRxViewModelFactory::class.java.isAssignableFrom(companionClass)) {
+    return declaredClasses.firstOrNull {
+        MvRxViewModelFactory::class.java.isAssignableFrom(it)
```

Comment:
afaik something like `filterInstanceOf` won't work because we need to match on the inner generic type of `class`